### ISSUE TITLE
Rating titles with whitespace results in broken ID attributes

### ISSUE THREAD

**Issue by chicgeek** [magento/magento2#5451]
The review form uses the rating labels to output element attributes in the FE [form.phtml](https://github.com/magento/magento2/blob/develop/app/code/Magento/Review/view/frontend/templates/form.phtml) template. These attribute values include IDs that should not include whitespace - this whitespace should be stripped out (likely with `str_replace`) for all (but one) instances of:

```
<?php echo $block->escapeHtml($_rating->getRatingCode()) ?>
```

The _only_ instance that should not have whitespace is the FE-visible label, the second instance on [L30](https://github.com/magento/magento2/blob/develop/app/code/Magento/Review/view/frontend/templates/form.phtml#L30):

```
<label class="label" id="<?php echo $block->escapeHtml($_rating->getRatingCode()) ?>_rating_label"><span><?php echo $block->escapeHtml($_rating->getRatingCode()) ?></span></label>
```
## Steps to reproduce

This issue is present in 2.0.6, 2.0.7, and 2.1 (and likely earlier versions).
1. In the admin panel, go to **Stores > Attribute > Rating**.
2. Click _Add new rating_.
3. Create new rating with a default value of `test with space`. Make it visible on all stores, enable it. Save.
4. Clear caches and view the markup on the frontend.
## Desired result

L30, with the first rating label whitespace replaced with non-whitespace characters:

``` html
<label class="label" id="test_with_space_rating_label"><span>test with space</span></label>
```
## Actual result

L30, as-is:

``` html
<label class="label" id="test with space_rating_label"><span>test with space</span></label>
```

This produces an element with multiple IDs. While this is valid markup, the likelihood of unintentionally having multiple elements with the _same_ ID (invalid markup) is fairly high.
## Note

Earlier versions may require the patch that is mentioned in this issue to make ratings visible to stores: https://github.com/magento/magento2/issues/4455

**Comment by AVoskoboinikov** [comment]
Hi @chicgeek 

Thanks for contributing. I've created internal ticket (#MAGETWO-58339) to fix this. This issue will be closed after fix.

Have a good day!

**Comment by magento-engcom-team** [comment]
@chicgeek, thank you for your report.
We've created internal ticket(s) MAGETWO-58339 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
Hi @chicgeek. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1119 by @nmalevanec in 2.2-develop branch
Related commit(s):
 - [9d5e253a0405c3172c487dea05148225af4e6b0f](https://github.com/magento/magento2/commit/9d5e253a0405c3172c487dea05148225af4e6b0f)
 - [38e2e171b68732d60608e55dcfc8950941c7bb52](https://github.com/magento/magento2/commit/38e2e171b68732d60608e55dcfc8950941c7bb52)

The fix will be available with the upcoming 2.2.4 release.
